tests/test_from_file.py

```diff
@@ -0,0 +1,128 @@
+            self.Config, ["--config-file", str((tmp_path / "config.json")), "--a", "7"], load_from_config=True
+        )
+        assert config.a == 7
+        assert config.b == "something"
+        assert config.c is False
+        assert config.z == "dummy"
+
+    def test_nested_with_file_and_override(self, tmp_path: Path) -> None:
+        (tmp_path / "config.yaml").write_text(dump({"a": 6, "s": {"a": 1, "b": "c", "s": {"q": "z"}}}))
+        config = parse(
+            self.Config, ["--config-file", str((tmp_path / "config.yaml")), "--s-b", "d"], load_from_config=True
+        )
+        assert config.a == 6
+        assert config.b == "something"
+        assert config.c is False
+        assert config.z == "dummy"
+        assert config.s.a == 1
+        assert config.s.b == "d"
+        assert config.s.s.q == "z"
+
+    def test_nested_with_prefixed_keys(self, tmp_path: Path) -> None:
+        (tmp_path / "config.yaml").write_text(dump({"a": 6, "s_a": 1, "s": {"b": "c"}}))
+        config = parse(self.Config, ["--config-file", str((tmp_path / "config.yaml"))], load_from_config=True)
+        assert config.a == 6
+        assert config.s.a == 1
+        assert config.s.b == "c"
+
+    def test_nested_with_duplicate_keys(self, tmp_path: Path) -> None:
+        (tmp_path / "config.yaml").write_text(dump({"a": 6, "s_a": 1, "s": {"a": 2, "b": "c"}}))
+        with raises(KeyError) as e:
+            parse(self.Config, ["--config-file", str((tmp_path / "config.yaml"))], load_from_config=True)
+        assert str(e.value) == "'Collision between keys in config file on key s_a.'"
+
+    def test_parse_with_nonexistent_file(self, tmp_path: Path) -> None:
+        with raises(FileNotFoundError):
+            parse(self.Config, ["--config-file", str((tmp_path / "config.json"))], load_from_config=True)
+
+    def test_parse_with_extra_fields(self, tmp_path: Path) -> None:
+        (tmp_path / "config.json").write_text(dumps({"a": 6, "d": "this_is_extra"}))
+        with warns(UserWarning) as warnings:
+            parse(self.Config, ["--config-file", str((tmp_path / "config.json"))], load_from_config=True)
+        assert (
+            str(warnings[0].message) == "The following keys from the provided configuration file were not consumed: d"
+        )
+
+
+class TestParseFromFileNoDefaults:
+    @dataclass
+    class Config:
+        a: int
+        b: str = field(metadata=dict(positional=True))
+        s: SubConfig = field(default_factory=SubConfig)
+        z: str = "dummy"
+
+    def test_parse_with_file_fields_remain_required(self, tmp_path: Path, capsys) -> None:
+        (tmp_path / "config.json").write_text(dumps({"a": 6, "b": "something"}))
+        with raises(SystemExit):
+            parse(self.Config, ["--config-file", str((tmp_path / "config.json"))], load_from_config=True)
+        captured = capsys.readouterr()
+        assert "the following arguments are required: --a, b" in captured.err
+
+
+class TestFileCollision:
```

Comment:
Not sure if this class matches the tests below, looks like it has to do with the file existing or not

Reply:
The test verifies that a field named `config_file`, which collides with the `--config-file` argument which is automatically added, can safely be added when loading from file is disabled, but raises an exception when it is enabled.